### PR TITLE
Fix ank apply flickering with unchanged table display

### PR DESCRIPTION
**Pull request by inf17101**
Issues: #250 

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: common/src/objects/workload_state.rs [hunk 1]
```diff
@@ -23,14 +23,15 @@ use crate::std_extensions::UnreachableOption;
...
-pub const NO_MORE_RETRIES_MSG: &str = "No more retries.";
+pub const NO_MORE_RETRIES_MSG: &str = "No more retries";
 
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
 pub enum PendingSubstate {
     Initial = 0,
     WaitingToStart = 1,
     Starting = 2,
     StartingFailed = 8,
+    RetryStarting = 9,
```
[the remark this answers is not in the document]
Comment (krucod3) [reply common/src/objects/workload_state.rs:34]: My initial idea was to add a new end state, but I like the solution to have starting failed kind of as an "end state" and just reduce the flickering by staying in the same state. 

If it is indeed a new state or not is a very good question. I actually cannot think of a situation in which the new state would be automatically evaluated. Maybe later we will make the retry configurable on system or even workload level, but that is a different story. 

@christoph-hamm, @inf17101, can you think of a use-case in which we need the "RetryStarting" state read from other workloads? If not, staying in "Starting" is probably the easiest way to go. 

The additional information with the counter is also really cool. If we stay in the "Starting" state, I would also keep the separate `retry_starting` call. This way we can add the state later if required.

[the remark this answers is not in the document]
Comment (inf17101) [reply common/src/objects/workload_state.rs:34]: Ok makes sense. As long as @christoph-hamm does not want find a use case for the extra sub-state, I will move the retires to the Pending(Starting) state but keep the additional info indicating that retries are done. This is then more flexible and less changes if we want to disable the retry handling. As final state I use already the Pending(StartingFailed) state which I will keep then.

[the remark this answers is not in the document]
Comment (inf17101) [reply common/src/objects/workload_state.rs:34]: It is now like that: 
image

**Comment by inf17101** [comment]
Closed because my PR #303 includes the wrap/truncate functionality in the table output on top.

**Pull request closed without merging**